### PR TITLE
Minor refactor for command suggestion logic

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Avoid side effects cased as part of #9783 

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
cc @deepu105

File: cli/utils.js [hunk 1]
```diff
@@ -88,12 +88,10 @@ const initHelp = (program, cliName) => {
     });
 
     program.on('command:*', name => {
-        logger.error(`${chalk.yellow(name)} is not a known command. See '${chalk.white(`${cliName} --help`)}'.`);
+        console.error(`${chalk.yellow(name)} is not a known command. See '${chalk.white(`${cliName} --help`)}'.`);
```
Comment (deepu105) [inline cli/utils.js]: please wrap entire text in `chalk.red(`

Comment (jamesgeorge007) [reply cli/utils.js]: :+1:

**Comment by jamesgeorge007** [comment]
@deepu105 Thoughts :thinking:

**Comment by jamesgeorge007** [comment]
cc @pascalgrimaud

**Pull request merged**